### ISSUE TITLE
SA Billing Error Report] dashboard#index (ActionView::Template::Error) "undefined method `company_name' for nil:NilClas...

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1892]
In GitLab by @gregory.robleski on Jun 24, 2020, 06:44

From: answernet@sabilling.com <answernet@sabilling.com>
Sent: Tuesday, June 23, 2020 2:16 PM
To: apperrors@sahosted.com
Subject: [SA Billing Error Report] dashboard#index (ActionView::Template::Error) "undefined method `company_name' for nil:NilClas...

An ActionView::Template::Error occurred in dashboard#index:

undefined method `company_name' for nil:NilClass
app/views/clients/dashboard/index.html.haml:19:in `_app_views_clients_dashboard_index_html_haml__2210993030108624896_55717800'

-------------------------------
Request:
-------------------------------

* URL : http://answernet.sabilling.com/clients
* HTTP Method: GET
* IP address : 24.52.126.10
* Parameters : {"action"=>"index", "controller"=>"clients/dashboard"}
* Timestamp : 06/23/2020 20:16
* Server : answernet.sabilling.com
* Rails root : /opt/sabilling
* Process: 10942

-------------------------------
Session:
-------------------------------

* session id: "342b1520f1a7a0df7adc579f37844472"
* data: {"session_id"=>"342b1520f1a7a0df7adc579f37844472",
"redirect_count"=>0,
"warden.user.client.key"=>
[["5ed12338cf85c15726001a6d"], "$2a$10$f5CCK/IzEi.WZXewfaK7p."],
"flash"=>
#<ActionDispatch::Flash::FlashHash:0x000000001dab6cc0
@closed=false,
@flashes={:alert=>"You are already signed in."},
@now=nil,
@used=#<Set: {:alert}>>}

-------------------------------
Environment:
-------------------------------

* HTTP_ACCEPT : text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9
* HTTP_ACCEPT_ENCODING : gzip, deflate, br
* HTTP_ACCEPT_LANGUAGE : en-US,en;q=0.9
* HTTP_CONNECTION : close
* HTTP_COOKIE : _billing_api_session=BAh7CUkiD3Nlc3Npb25faWQGOgZFVEkiJTM0MmIxNTIwZjFhN2EwZGY3YWRjNTc5ZjM3ODQ0NDcyBjsAVEkiE3JlZGlyZWN0X2NvdW50BjsARmkASSIbd2FyZGVuLnVzZXIuY2xpZW50LmtleQY7AFRbB1sGVToaTW9wZWQ6OkJTT046Ok9iamVjdElkIhFe0SM4z4XBVyYAGm1JIiIkMmEkMTAkZjVDQ0svSXpFaS5XWlhld2ZhSzdwLgY7AFRJIgpmbGFzaAY7AFRvOiVBY3Rpb25EaXNwYXRjaDo6Rmxhc2g6OkZsYXNoSGFzaAk6CkB1c2VkbzoIU2V0BjoKQGhhc2h9AEY6DEBjbG9zZWRGOg1AZmxhc2hlc3sGOgphbGVydEkiH1lvdSBhcmUgYWxyZWFkeSBzaWduZWQgaW4uBjsAVDoJQG5vdzA%3D--11236c9bb833527531e4a42d4c8d1951eca99489
* HTTP_HOST : answernet.sabilling.com
* HTTP_REFERER : https://answernet.com/login/
* HTTP_SEC_FETCH_DEST : document
* HTTP_SEC_FETCH_MODE : navigate
* HTTP_SEC_FETCH_SITE : cross-site
* HTTP_SEC_FETCH_USER : ?1
* HTTP_UPGRADE_INSECURE_REQUESTS : 1
* HTTP_USER_AGENT : Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Safari/537.36
* HTTP_VERSION : HTTP/1.0
* HTTP_X_FORWARDED_FOR : 24.52.126.10
* HTTP_X_REAL_IP : 24.52.126.10
* ORIGINAL_FULLPATH : /clients/
* PATH_INFO : /clients
* QUERY_STRING :
* REMOTE_ADDR : 127.0.0.1
* REQUEST_METHOD : GET
* REQUEST_PATH : /clients/
* REQUEST_URI : /clients/
* SCRIPT_NAME :
* SERVER_NAME : answernet.sabilling.com
* SERVER_PORT : 80
* SERVER_PROTOCOL : HTTP/1.0
* SERVER_SOFTWARE : Unicorn 5.4.0
* action_controller.instance : #<Clients::DashboardController:0x000000001dab4fb0>
* action_dispatch.backtrace_cleaner : #<Rails::BacktraceCleaner:0x0000000004f91f88>
* action_dispatch.cookies : #<ActionDispatch::Cookies::CookieJar:0x000000001dab7a08>
* action_dispatch.logger : #<ActiveSupport::TaggedLogging:0x00000000057c86c0>
* action_dispatch.parameter_filter : [:password, :ccNum, :ccCV2, :ccExpMonth, :ccExpYear, :check_acct, /RAW_POST_DATA/, /RAW_POST_DATA/, /RAW_POST_DATA/]
* action_dispatch.remote_ip : 24.52.126.10
* action_dispatch.request.content_type :
* action_dispatch.request.formats : [#<Mime::Type:0x000000000284d9b8 @synonyms=["application/xhtml+xml"], @symbol=:html, @string="text/html", @hash=4240037049767547472>]
* action_dispatch.request.parameters : {"action"=>"index", "controller"=>"clients/dashboard"}
* action_dispatch.request.path_parameters : {:action=>"index", :controller=>"clients/dashboard"}
* action_dispatch.request.query_parameters : {}
* action_dispatch.request.request_parameters : {}
* action_dispatch.request.unsigned_session_cookie: {"session_id"=>"342b1520f1a7a0df7adc579f37844472", "redirect_count"=>0, "warden.user.client.key"=>[["5ed12338cf85c15726001a6d"], "$2a$10$f5CCK/IzEi.WZXewfaK7p."], "flash"=>#<ActionDispatch::Flash::FlashHash:0x000000001dab6cc0 @used=#<Set: {:alert}>, @closed=false, @flashes={:alert=>"You are already signed in."}, @now=nil>}
* action_dispatch.request_id : 82c35aa3a4995fb077211abb3c442fe3
* action_dispatch.routes : #<ActionDispatch::Routing::RouteSet:0x0000000005762d98>
* action_dispatch.secret_token : 51106aba165bc2293b002675ccb821c83c7a88aa71acad754f8e473b8fca8b890ee4ef2f9b2ef5bf0b269c4a07cbe47f8f6a65925cd333701696dadbd2d93252
* action_dispatch.show_detailed_exceptions : false
* action_dispatch.show_exceptions : true
* rack.errors : #<File:0x0000000001040960>
* rack.hijack : #<Unicorn::HttpParser:0x00000000019c62c8>
* rack.hijack? : true
* rack.input : #<StringIO:0x0000000001afc2a0>
* rack.logger : #<Logger:0x0000000001afe640>
* rack.multiprocess : true
* rack.multithread : false
* rack.request.cookie_hash : {"_billing_api_session"=>"BAh7CUkiD3Nlc3Npb25faWQGOgZFVEkiJTM0MmIxNTIwZjFhN2EwZGY3YWRjNTc5ZjM3ODQ0NDcyBjsAVEkiE3JlZGlyZWN0X2NvdW50BjsARmkASSIbd2FyZGVuLnVzZXIuY2xpZW50LmtleQY7AFRbB1sGVToaTW9wZWQ6OkJTT046Ok9iamVjdElkIhFe0SM4z4XBVyYAGm1JIiIkMmEkMTAkZjVDQ0svSXpFaS5XWlhld2ZhSzdwLgY7AFRJIgpmbGFzaAY7AFRvOiVBY3Rpb25EaXNwYXRjaDo6Rmxhc2g6OkZsYXNoSGFzaAk6CkB1c2VkbzoIU2V0BjoKQGhhc2h9AEY6DEBjbG9zZWRGOg1AZmxhc2hlc3sGOgphbGVydEkiH1lvdSBhcmUgYWxyZWFkeSBzaWduZWQgaW4uBjsAVDoJQG5vdzA=--11236c9bb833527531e4a42d4c8d1951eca99489"}
* rack.request.cookie_string : _billing_api_session=BAh7CUkiD3Nlc3Npb25faWQGOgZFVEkiJTM0MmIxNTIwZjFhN2EwZGY3YWRjNTc5ZjM3ODQ0NDcyBjsAVEkiE3JlZGlyZWN0X2NvdW50BjsARmkASSIbd2FyZGVuLnVzZXIuY2xpZW50LmtleQY7AFRbB1sGVToaTW9wZWQ6OkJTT046Ok9iamVjdElkIhFe0SM4z4XBVyYAGm1JIiIkMmEkMTAkZjVDQ0svSXpFaS5XWlhld2ZhSzdwLgY7AFRJIgpmbGFzaAY7AFRvOiVBY3Rpb25EaXNwYXRjaDo6Rmxhc2g6OkZsYXNoSGFzaAk6CkB1c2VkbzoIU2V0BjoKQGhhc2h9AEY6DEBjbG9zZWRGOg1AZmxhc2hlc3sGOgphbGVydEkiH1lvdSBhcmUgYWxyZWFkeSBzaWduZWQgaW4uBjsAVDoJQG5vdzA%3D--11236c9bb833527531e4a42d4c8d1951eca99489
* rack.request.query_hash : {}
* rack.request.query_string :
* rack.run_once : false
* rack.session : {"session_id"=>"342b1520f1a7a0df7adc579f37844472", "redirect_count"=>0, "warden.user.client.key"=>[["5ed12338cf85c15726001a6d"], "$2a$10$f5CCK/IzEi.WZXewfaK7p."], "flash"=>#<ActionDispatch::Flash::FlashHash:0x000000001dab6cc0 @used=#<Set: {:alert}>, @closed=false, @flashes={:alert=>"You are already signed in."}, @now=nil>}
* rack.session.options : {:path=>"/", :domain=>nil, :expire_after=>nil, :secure=>false, :httponly=>true, :defer=>false, :renew=>false, :secret=>"2c9fb683db914d76dfb0cf26238c4105db17ae4ee5bc7dd10ebb849eaa57", :coder=>#<Rack::Session::Cookie::Base64::Marshal:0x000000000582e5d8>, :id=>"342b1520f1a7a0df7adc579f37844472"}
* rack.url_scheme : http
* rack.version : [1, 2]
* unicorn.socket : #<Kgio::Socket:0x000000001b793888>
* warden : Warden::Proxy:248886300 @config={:default_scope=>:client, :scope_defaults=>{}, :default_strategies=>{:client=>[:rememberable, :database_authenticatable], :user=>[:local_override, :ldap_authenticatable]}, :intercept_401=>false, :failure_app=>#<Devise::Delegator:0x000000000515f5e0>}

-------------------------------
Backtrace:
-------------------------------

app/views/clients/dashboard/index.html.haml:19:in `_app_views_clients_dashboard_index_html_haml__2210993030108624896_55717800'
actionpack (3.2.22.5) lib/action_view/template.rb:145:in `block in render'
activesupport (3.2.22.5) lib/active_support/notifications.rb:125:in `instrument'
actionpack (3.2.22.5) lib/action_view/template.rb:143:in `render'
actionpack (3.2.22.5) lib/action_view/renderer/template_renderer.rb:48:in `block (2 levels) in render_template'
actionpack (3.2.22.5) lib/action_view/renderer/abstract_renderer.rb:38:in `block in instrument'
activesupport (3.2.22.5) lib/active_support/notifications.rb:123:in `block in instrument'
activesupport (3.2.22.5) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (3.2.22.5) lib/active_support/notifications.rb:123:in `instrument'
actionpack (3.2.22.5) lib/action_view/renderer/abstract_renderer.rb:38:in `instrument'
actionpack (3.2.22.5) lib/action_view/renderer/template_renderer.rb:47:in `block in render_template'
actionpack (3.2.22.5) lib/action_view/renderer/template_renderer.rb:55:in `render_with_layout'
actionpack (3.2.22.5) lib/action_view/renderer/template_renderer.rb:46:in `render_template'
actionpack (3.2.22.5) lib/action_view/renderer/template_renderer.rb:18:in `render'
actionpack (3.2.22.5) lib/action_view/renderer/renderer.rb:43:in `render_template'
actionpack (3.2.22.5) lib/action_view/renderer/renderer.rb:24:in `render'
actionpack (3.2.22.5) lib/abstract_controller/rendering.rb:111:in `_render_template'
actionpack (3.2.22.5) lib/action_controller/metal/streaming.rb:225:in `_render_template'
actionpack (3.2.22.5) lib/abstract_controller/rendering.rb:104:in `render_to_body'
actionpack (3.2.22.5) lib/action_controller/metal/renderers.rb:28:in `render_to_body'
actionpack (3.2.22.5) lib/action_controller/metal/compatibility.rb:50:in `render_to_body'
actionpack (3.2.22.5) lib/abstract_controller/rendering.rb:89:in `render'
actionpack (3.2.22.5) lib/action_controller/metal/rendering.rb:16:in `render'
actionpack (3.2.22.5) lib/action_controller/metal/instrumentation.rb:40:in `block (2 levels) in render'
activesupport (3.2.22.5) lib/active_support/core_ext/benchmark.rb:5:in `block in ms'
/usr/local/rvm/rubies/ruby-2.5.0/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
activesupport (3.2.22.5) lib/active_support/core_ext/benchmark.rb:5:in `ms'
actionpack (3.2.22.5) lib/action_controller/metal/instrumentation.rb:40:in `block in render'
actionpack (3.2.22.5) lib/action_controller/metal/instrumentation.rb:83:in `cleanup_view_runtime'
actionpack (3.2.22.5) lib/action_controller/metal/instrumentation.rb:39:in `render'
wicked_pdf (1.1.0) lib/wicked_pdf/pdf_helper.rb:42:in `render_with_wicked_pdf'
actionpack (3.2.22.5) lib/action_controller/metal/implicit_render.rb:10:in `default_render'
actionpack (3.2.22.5) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
actionpack (3.2.22.5) lib/abstract_controller/base.rb:167:in `process_action'
actionpack (3.2.22.5) lib/action_controller/metal/rendering.rb:10:in `process_action'
actionpack (3.2.22.5) lib/abstract_controller/callbacks.rb:18:in `block in process_action'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:447:in `_run__2233250473209944089__process_action__3152935203223614713__callbacks'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:405:in `__run_callback'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:385:in `_run_process_action_callbacks'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (3.2.22.5) lib/abstract_controller/callbacks.rb:17:in `process_action'
actionpack (3.2.22.5) lib/action_controller/metal/rescue.rb:29:in `process_action'
actionpack (3.2.22.5) lib/action_controller/metal/instrumentation.rb:30:in `block in process_action'
activesupport (3.2.22.5) lib/active_support/notifications.rb:123:in `block in instrument'
activesupport (3.2.22.5) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
activesupport (3.2.22.5) lib/active_support/notifications.rb:123:in `instrument'
actionpack (3.2.22.5) lib/action_controller/metal/instrumentation.rb:29:in `process_action'
actionpack (3.2.22.5) lib/action_controller/metal/params_wrapper.rb:207:in `process_action'
actionpack (3.2.22.5) lib/abstract_controller/base.rb:121:in `process'
actionpack (3.2.22.5) lib/abstract_controller/rendering.rb:46:in `process'
actionpack (3.2.22.5) lib/action_controller/metal.rb:203:in `dispatch'
actionpack (3.2.22.5) lib/action_controller/metal/rack_delegation.rb:14:in `dispatch'
actionpack (3.2.22.5) lib/action_controller/metal.rb:246:in `block in action'
actionpack (3.2.22.5) lib/action_dispatch/routing/route_set.rb:73:in `dispatch'
actionpack (3.2.22.5) lib/action_dispatch/routing/route_set.rb:36:in `call'
journey (1.0.4) lib/journey/router.rb:68:in `block in call'
journey (1.0.4) lib/journey/router.rb:56:in `each'
journey (1.0.4) lib/journey/router.rb:56:in `call'
actionpack (3.2.22.5) lib/action_dispatch/routing/route_set.rb:608:in `call'
mongoid (3.1.7) lib/rack/mongoid/middleware/identity_map.rb:34:in `block in call'
mongoid (3.1.7) lib/mongoid/unit_of_work.rb:39:in `unit_of_work'
mongoid (3.1.7) lib/rack/mongoid/middleware/identity_map.rb:34:in `call'
exception_notification (4.1.1) lib/exception_notification/rack.rb:32:in `call'
warden (1.2.7) lib/warden/manager.rb:36:in `block in call'
warden (1.2.7) lib/warden/manager.rb:35:in `catch'
warden (1.2.7) lib/warden/manager.rb:35:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/best_standards_support.rb:17:in `call'
rack (1.4.7) lib/rack/etag.rb:23:in `call'
rack (1.4.7) lib/rack/conditionalget.rb:25:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/head.rb:14:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/params_parser.rb:21:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/flash.rb:242:in `call'
rack (1.4.7) lib/rack/session/abstract/id.rb:210:in `context'
rack (1.4.7) lib/rack/session/abstract/id.rb:205:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/cookies.rb:341:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:405:in `_run__1429002392000842749__call__1060899571951328473__callbacks'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:405:in `__run_callback'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:385:in `_run_call_callbacks'
activesupport (3.2.22.5) lib/active_support/callbacks.rb:81:in `run_callbacks'
actionpack (3.2.22.5) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/reloader.rb:65:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/remote_ip.rb:31:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/debug_exceptions.rb:16:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/show_exceptions.rb:56:in `call'
railties (3.2.22.5) lib/rails/rack/logger.rb:32:in `call_app'
railties (3.2.22.5) lib/rails/rack/logger.rb:16:in `block in call'
activesupport (3.2.22.5) lib/active_support/tagged_logging.rb:22:in `tagged'
railties (3.2.22.5) lib/rails/rack/logger.rb:16:in `call'
request_store (1.4.1) lib/request_store/middleware.rb:19:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/request_id.rb:22:in `call'
rack (1.4.7) lib/rack/methodoverride.rb:21:in `call'
rack (1.4.7) lib/rack/runtime.rb:17:in `call'
activesupport (3.2.22.5) lib/active_support/cache/strategy/local_cache.rb:72:in `call'
rack (1.4.7) lib/rack/lock.rb:15:in `call'
actionpack (3.2.22.5) lib/action_dispatch/middleware/static.rb:83:in `call'
railties (3.2.22.5) lib/rails/engine.rb:484:in `call'
railties (3.2.22.5) lib/rails/application.rb:231:in `call'
railties (3.2.22.5) lib/rails/railtie/configurable.rb:30:in `method_missing'
unicorn (5.4.0) lib/unicorn/http_server.rb:606:in `process_client'
unicorn (5.4.0) lib/unicorn/http_server.rb:701:in `worker_loop'
unicorn (5.4.0) lib/unicorn/http_server.rb:549:in `spawn_missing_workers'
unicorn (5.4.0) lib/unicorn/http_server.rb:142:in `start'
unicorn (5.4.0) bin/unicorn:126:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.5.0/bin/unicorn:23:in `load'
/usr/local/rvm/gems/ruby-2.5.0/bin/unicorn:23:in `<main>'
/usr/local/rvm/gems/ruby-2.5.0/bin/ruby_executable_hooks:24:in `eval'
/usr/local/rvm/gems/ruby-2.5.0/bin/ruby_executable_hooks:24:in `<main>'

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 24, 2020, 23:19

assigned to @amishra and unassigned @ranand

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 25, 2020, 24:44

@kdjstudios @Robleski 

Thank you for the update.

We looked into this and our findings are described below: 

* User jeffrey.casey changed the customer on 25th may 2020.
* we have one script if there is no account for the customer then we delete the customer.
* and now, there is no customer associate for the client because the customer is deleted. 



![Screenshot_from_2020-06-25_11-51-28](/uploads/23ad62a2720c3324bd688efa5ec8324d/Screenshot_from_2020-06-25_11-51-28.png)

**In short:**

* The client has no customer and as per the current functionality of SAB unable to handle this scenario.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2020, 08:09

@amishra 

Are you saying we allowed for a user to save an account without a customer? If so, that seems odd since that is a required field and we have validation to check for that before allowing that to be saved, Right? If not, we should have as there has been multiple tickets addressing validation and required fields. If this is also correct, please create a place holder customer so the account can be viewed again in the UI and that we can instruct operations to handle this correctly. 

If that above is not correct. Please explain in further detail.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 25, 2020, 08:51

This was reviewed on the phone call today. This occurs due to a customer being deleted and it's associated client portal login not also being deleted. Please manually delete the client portal logins for these customers, and then we will need to develop/update the delete script to also delete a customer's client portal login.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 26, 2020, 05:06

@kdjstudios @Robleski 

Thank you for the update.

We have two(2) scenarios while deleting a client. Please have a look and let us know if we need to change anything.

**Customers and client lists:**
* Customer Cus1
* Customer Cus2
* Customer Cus3
* Client C1

`scenario 1`: If the client (C1) has a **Cus1** customer and the customer gets deleted. **IN THIS CASE: Client also should be deleted.**
  
`scenario 2`: If the client (C1) has **Cus1** and **Cus2** customers and customer **Cus1** get deleted. **IN THIS CASE: Client also should `NOT` be deleted because the client has already one customer(**Cus2**) assign.**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 26, 2020, 10:15

@amishra @Robleski 

Very great question that is something I had not thought about since we just developed the parent/child customer Portal logins. 

**Scenario 1:** is your basic 1 to 1 setup (most common). where there is Customer A who has their Portal A. When Customer A does not have any more accounts underneath it, then as part of the clean up process it should also delete their Portal A too. 

**Scenario 2:** is where we have a parent Customer A with child customers B and C who has their Portal A. These would result in two different types of processing
* If child Customer B no longer have any accounts under them, then the clean up process it should not delete the Portal A, but it should remove the Link to Customer A or Portal A (however that is linked via the code). Portal A will still exist and now only have access to Customer A and Customer C.
* If parent customer A no longer has any accounts under them, then the clean up process should delete the Portal A too. This should also result in Customer B and C to no longer have a Portal association. 

Does that make sense? or is there further explanation needed?

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jun 30, 2020, 04:17

@kdjstudios  @Robleski 

Thanks for the update.

We have completed the development. But we can't delete the customer from UI.So I just want to confirm do you want to test this?
 
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 30, 2020, 09:39

@rranjan @Robleski 

Are you not able to test this by moving the accounts from under a customer and then having the script delete the customer?

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on Jul 1, 2020, 08:14

@kdjstudios  @Robleski 

>Are you not able to test this by moving the accounts from under a customer and then having the script delete the customer?

We have tested it on our end if you want to test it then we need to set cronjob at a certain time so that you can test it.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 1, 2020, 09:01

@rranjan @Robleski 

Thank you for the update. If you have tested this then it should be good. Marking for ready.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jul 11, 2020, 03:39

@kdjstudios @Robleski 

We have successfully released on both of the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jul 13, 2020, 10:22

closed
